java/ingest/src/main/java/sleeper/ingest/IngestRecordsUsingPropertiesSpecifiedMethod.java

```diff
@@ -124,30 +108,17 @@ public static List<FileInfo> ingestFromRecordIterator(
             totalArrowBytesRequired = instanceProperties.getLong(ARROW_INGEST_WORKING_BUFFER_BYTES) +
                     instanceProperties.getLong(ARROW_INGEST_BATCH_BUFFER_BYTES);
         }
-        try (BufferAllocator arrowBufferAllocator =
-                     (totalArrowBytesRequired > 0) ?
-                             ((bufferAllocator == null) ?
-                                     new RootAllocator(totalArrowBytesRequired) :
-                                     bufferAllocator.newChildAllocator("Ingest buffer", totalArrowBytesRequired, totalArrowBytesRequired)) :
-                             null;
-             // Create an IngestRecords object
-             IngestCoordinator<Record> ingestCoordinator = createIngestCoordinatorFromProperties(
-                     objectFactory,
-                     sleeperStateStore,
-                     instanceProperties,
-                     tableProperties,
-                     localWorkingDirectory,
-                     arrowBufferAllocator,
-                     internalS3AsyncClient,
-                     internalHadoopConfiguration,
-                     sleeperIteratorClassName,
-                     sleeperIteratorConfig)) {
-            // Write all the records to the IngestCoordinator
-            while (recordIterator.hasNext()) {
-                ingestCoordinator.write(recordIterator.next());
-            }
-            return ingestCoordinator.closeReturningFileInfoList();
+        try (BufferAllocator arrowBufferAllocator = (totalArrowBytesRequired > 0) ? new RootAllocator(totalArrowBytesRequired) : null) {
```

Comment:
There was a try/finally around the IngestCoordinator before, there isn't one any more. I think we'll need to build the IngestCoordinator separately and do the IngestRecordsFromIterator bit inside the try/finally.

Reply:
Yup, looks like you sorted it.